src/tiny_gea2_single_wire_interface.c

```diff
@@ -0,0 +1,647 @@
+  data_length_bytes_not_included_in_data = tiny_gea3_packet_transmission_overhead - tiny_gea3_packet_overhead,
+  crc_size = sizeof(uint16_t),
+  packet_bytes_not_included_in_payload = crc_size + offsetof(tiny_gea3_packet_t, payload),
+  unbuffered_bytes = 2 // STX, ETX
+};
+
+enum {
+  send_state_data,
+  send_state_crc_msb,
+  send_state_crc_lsb,
+  send_state_etx,
+  send_state_stx,
+  send_state_done,
+};
+
+static void state_idle(tiny_fsm_t* fsm, const tiny_fsm_signal_t signal, const void* data);
+static void state_receive(tiny_fsm_t* fsm, const tiny_fsm_signal_t signal, const void* data);
+static void state_idle_cooldown(tiny_fsm_t* fsm, const tiny_fsm_signal_t signal, const void* data);
+static void state_send(tiny_fsm_t* fsm, const tiny_fsm_signal_t signal, const void* data);
+static void state_wait_for_ack(tiny_fsm_t* fsm, const tiny_fsm_signal_t signal, const void* data);
+static void state_collision_cooldown(tiny_fsm_t* fsm, const tiny_fsm_signal_t signal, const void* data);
+
+static self_t* interface_from_fsm(tiny_fsm_t* fsm)
+{
+  return container_of(self_t, _private.fsm, fsm);
+}
+
+static void byte_received(void* context, const void* _args)
+{
+  reinterpret(instance, context, self_t*);
+  reinterpret(args, _args, const tiny_uart_on_receive_args_t*);
+
+  tiny_fsm_send_signal(&instance->_private.fsm, signal_byte_received, &args->byte);
+}
+
+#define needs_escape(_byte) ((_byte & 0xFC) == tiny_gea3_esc)
+
+#define is_broadcast_address(_address) ((gea2_broadcast_mask & _address) == gea2_broadcast_mask)
+
+static void state_idle(tiny_fsm_t* fsm, const tiny_fsm_signal_t signal, const void* data)
+{
+  self_t* instance = interface_from_fsm(fsm);
+  switch(signal) {
+    case tiny_fsm_signal_entry:
+    case signal_send_ready:
+      if(instance->_private.send.active) {
+        tiny_fsm_transition(fsm, state_send);
+      }
+      break;
+
+    case signal_byte_received: {
+      reinterpret(byte, data, const uint8_t*);
+
+      if(instance->_private.receive.packet_ready) {
+        break;
+      }
+
+      if(*byte == tiny_gea3_stx && !instance->_private.receive.packet_ready) {
+        tiny_fsm_transition(fsm, state_receive);
+      }
+      else {
+        tiny_fsm_transition(fsm, state_idle_cooldown);
+      }
+    } break;
+  }
+}
+
+static void reflection_timeout(void* context)
+{
+  self_t* instance = context;
+  tiny_fsm_send_signal(&instance->_private.fsm, signal_reflection_timeout, NULL);
+}
+
+static bool determine_byte_to_send_considering_escapes(self_t* instance, uint8_t byte, uint8_t* byteTosend)
+{
+  if(!instance->_private.send.escaped && needs_escape(byte)) {
+    instance->_private.send.escaped = true;
+    *byteTosend = tiny_gea3_esc;
+  }
+  else {
+    instance->_private.send.escaped = false;
+    *byteTosend = byte;
+  }
+
+  return !instance->_private.send.escaped;
+}
+
+static void send_next_byte(self_t* instance)
+{
+  uint8_t byteTosend = 0;
+
+  tiny_timer_start(
```

Comment:
I noticed that this timer is meant to signal that there was a failure in the send but it doesn't seem like anything stops the timer if the next byte is sent. Is there assumption that there will always be failture?